### ISSUE TITLE
Potentially fix how we read APP_VARIANT env var in app code?

### ISSUE THREAD

**Issue by achou11** [digidem/comapeo-mobile#597]
See #596 

Based on the docs, my understanding is that any env variables that are read in **app code** should be prefixed by `EXPO_PUBLIC_` in order to be properly inlined when bundling. There are a couple of places where we read `process.env.APP_VARIANT` in the app code and based on testing while in local development, it seems to read the value from `.env`, which is unexpected to me based on the documentation.

https://docs.expo.dev/build-reference/variables/
https://docs.expo.dev/guides/environment-variables/

Currently, things seem to work fine but I'm a little worried that this could be something that creates issues when creating builds with EAS

**Comment by achou11** [comment]
part of me wonders if this is causing occasional issues while in development where app code checks that reference the `APP_VARIANT` don't work as expected 🤔

**Comment by gmaclennan** [comment]
Yes, my understanding of how expo inlines environment variables means that `process.env.APP_VARIANT` is not available at runtime. Currently this is only used in app code to add extra debugging info [here](https://github.com/digidem/comapeo-mobile/blob/b423f24ff79167e190ae11eaaad48ee77b94df37/src/frontend/App.tsx#L24-L25) and [here](https://github.com/digidem/comapeo-mobile/blob/b423f24ff79167e190ae11eaaad48ee77b94df37/src/frontend/lib/intl.ts#L32). The consequence is minimal - these statements will never evaluate to `true` (because it is undefined), so extra debugging will never happen.

This is low-priority because it only affects dev builds, but we should definitely fix if we use `process.env.APP_VARIANT` for anything else.

**Comment by gmaclennan** [comment]
I had missed that we are using the babel plugin [`transform-inline-environment-variables`](https://github.com/digidem/comapeo-mobile/blob/develop/babel.config.js#L8) (SWM added it). This will be making it work.

I think we should use that babel transform over expo env variables, because it's more flexible, but we should specify the env variables we want to include, to avoid the small security risk of randoms creating PRs that read private env variables.
